.github/workflows/terraform-compatibility-matrix.yml

```diff
@@ -0,0 +1,50 @@
+name: "HashiCorp Terraform Compatibility Matrix"
+run-name: 'HashiCorp Terraform Compatibility Matrix ${{ inputs.atlas_cloud_env }}'
+
+on:
+  schedule:
+    - cron: "0 0 1-7 * 6" # runs first Saturday of the month at midnight UTC
+  workflow_dispatch:
+    inputs:
+      atlas_cloud_env:
+        description: 'Atlas cloud environment used, can be either `dev` or `qa`, empty for `dev`. Migration tests will always use `dev`'     
+        type: string
+        required: false
+      test_group:
+        description: 'Test group to run, e.g. advanced_cluster, empty for all'
+        type: string
+        required: false
+
+jobs:
+  get-supported-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - name: Get HashiCorp Terraform supported versions
+        shell: bash
+        id: get-terraform-supported-versions
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
is this needed?

Reply:
yes, GitHub API has rate limiting and with the github token limits are increased